App/Containers/RenameGroupModal.tsx

```diff
@@ -0,0 +1,152 @@
+import { groupActions } from '../features/group'
+
+import styles from './Styles/RenameGroup'
+import { color } from '../styles'
+
+interface StateProps {
+  groupName: string
+  renaming: boolean
+}
+
+interface DispatchProps {
+  rename: (newName: string) => void
+}
+
+interface ScreenProps {
+  cancel: () => void
+  complete: () => void
+}
+
+interface ModalProps {
+  isVisible: boolean
+}
+
+interface State {
+  newName: string,
+  startedRename: false
+}
+
+class RenameGroupScreen extends React.Component<ScreenProps & StateProps & DispatchProps, State> {
+
+  constructor(props: Props) {
+    super(props)
+    this.state = {
+      newName: props.groupName,
+      startedRename: false
+    }
+  }
+
+  componentDidUpdate() {
+    if (this.state.startedRename && !this.props.renaming) {
+      this.setState({
+        startedRename: false
+      })
+      this.props.complete()
+    }
+  }
+
+  render() {
+    const groupName = this.props.groupName
+    return (
+      <View style={styles.container}>
+        <Input
+          style={styles.inputStyle}
+          value={this.state.newName}
+          label={this.state.newName === '' ? 'Change the group name' : ''}
+          onChangeText={this.handleNewText}
+        />
+        <View style={styles.buttons}>
+          <TouchableOpacity
+            style={styles.buttonContainer}
+            onPress={this.props.cancel}
+          >
+            <Text style={styles.cancelButtonText}>Cancel</Text>
+          </TouchableOpacity>
+          <TouchableOpacity
+            style={styles.buttonContainer}
+            disabled={this.state.newName === '' || this.state.renaming}
+            onPress={this.rename}
+          >
+            <Text style={styles.confirmButtonText}>Rename</Text>
+          </TouchableOpacity>
+        </View>
+      </View>
+    )
+  }
+
+  handleNewText = (text: string) => {
+    this.setState({
+      newName: text
+    })
+  }
+
+  rename = () => {
+    this.props.rename(this.state.newName)
+    this.setState({
+      startedRename: true
+    })
+  }
+}
+
+const mapStateToProps = (state: RootState, ownProps: NavigationScreenProps<NavProps>): StateProps => {
+  const threadId = ownProps.threadId
+  const threadData = state.photoViewing.threads[threadId]
+  const groupName = threadData ? threadData.name : 'Unknown'
+  const renaming = Object.keys(state.group.renameGroup).indexOf(threadId) > -1
+  return {
+    groupName,
+    renaming
+  }
+}
+
+const mapDispatchToProps = (dispatch: Dispatch<RootAction>, ownProps: NavigationScreenProps<NavProps>) : DispatchProps => {
+  const threadId = ownProps.threadId
+  return {
+    rename: (newName: string) => { dispatch(groupActions.renameGroup.renameGroup.request({ threadId, name: newName })) }
+  }
+}
+
+export const RenameGroupComponent = connect(mapStateToProps, mapDispatchToProps)(RenameGroupScreen)
+
+export default class RenameGroupModal extends React.Component<ScreenProps & ModalProps> {
```

Comment:
Not really seeing the need for a while class to wrap the modal. You can just put this in the `render` of `RenameGroupScreen`